### PR TITLE
Mkulakow/model control api single model

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

**Review by mzegla — APPROVED** [review]

File: src/test/http_rest_api_handler_test.cpp [hunk 1]
```diff
@@ -56,6 +56,12 @@ class ConfigApi : public TestWithTempDir {
     void RemoveConfig() {
         std::filesystem::remove(configFilePath);
     }
+
+    void SetUpSingleModel(std::string modelPath, std::string modelName) {
+        char* n_argv[] = {"ovms", "--model_path", &modelPath[0], "--model_name", &modelName[0], "--file_system_poll_wait_seconds", "0"};
```
Comment (dkalinowski) [inline src/test/http_rest_api_handler_test.cpp:61]: modelPath.data() or modelPath.c_str() to get pointer of underlying char array

File: src/test/http_rest_api_handler_test.cpp [hunk 2]
```diff
@@ -56,6 +56,12 @@ class ConfigApi : public TestWithTempDir {
     void RemoveConfig() {
         std::filesystem::remove(configFilePath);
     }
+
+    void SetUpSingleModel(std::string modelPath, std::string modelName) {
```
Comment (dkalinowski) [inline src/test/http_rest_api_handler_test.cpp:60]: Pass by const reference

File: src/test/http_rest_api_handler_test.cpp (hunk 1 above)
Comment (dkalinowski) [reply src/test/http_rest_api_handler_test.cpp:61]: Are you sure that buffers do not need to exist after parsing? Lifetime of modelPath/modelName ends at the end oof SetUpSingleModel function

File: src/test/http_rest_api_handler_test.cpp [hunk 3]
```diff
@@ -128,6 +134,36 @@ TEST_F(ConfigReload, startWith1DummyThenReload) {
     EXPECT_EQ(status, ovms::StatusCode::OK_RELOADED);
 }
 
+TEST_F(ConfigReload, singleModel) {
+    ModelManagerTest manager;
+    SetUpSingleModel("/ovms/src/test/dummy", "dummy");
+    manager.startFromConfig();
```
Comment (dkalinowski) [inline src/test/http_rest_api_handler_test.cpp:140]: This is single model test but you call `startFromConfig` which is called by OVMS when started from config.jon

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**